src/services/fileServices/MdPageServices/MediaFileService.js

```diff
@@ -21,7 +21,18 @@ class MediaFileService {
 
   async create(sessionData, { fileName, directoryName, content }) {
     this.mediaNameChecks({ directoryName, fileName })
-    const sanitizedContent = await validateAndSanitizeFileUpload(content)
+
+    const [schema, fileContent] = content.split(",")
+    const fileBuffer = Buffer.from(fileContent, "base64")
+
+    //Scan file for virus - cloudmersive API
+    const virusScanRes = await scanFileForVirus(fileBuffer)
```

Comment:
could i ask why we split this from `validateAndSanitizeFileUpload`? placing it there seems to make sense actually

Reply:
Sure, thanks for asking. I'll explain my thoughts in two aspects here: 

1. The functionalities of these two file checks are fundamentally different. `validateAndSanitizeFileUpload` is to validate the file type & extension, and sanitize the file content. `scanFileForVirus` is to perform a post call to [Cloudmersive API endpoint ](https://api.cloudmersive.com/docs/virus.asp)to scan the file for virus. Depending on the use case, these two functions can be used independently and separately. For example, if a file uploading feature doesn't care about the file type & extension, but only want to perform virus scan, it can invoke the scanFileForVirus function and complete the job. Therefore I think logically it's better to split them into two different re-usable functions than combining them together. 
2. The way those methods were implemented are also different. `validateAndSanitizeFileUpload` is doing synchronous local check & purify work for the file content, while `scanFileForVirus` is doing a remote API call to [Cloudmersive ](https://api.cloudmersive.com/docs/virus.asp)using its sdk which requires a callback function to get the results. Therefore I need to use a Promise and await the result from the caller function. Combining them might make it challenging to enforce the order of the checks, regardless of what order we prefer. Hope that makes sense.

Reply:
I absolutely agree with you for the type script part. But I think these two functions should be in the same `file-upload-utils` file since they are all related and used by file uploading functionality. This file, together with all the other `js` files under `utils` folder should all be migrated to `ts` file in the future for better type checking/enforcement. 
But it would likely to be a separate effort to refactor the codebase which is beyond the scope of this PR. We will plan for a proper `Typescript` migration for the entire codebase (or at least the `util` folder).  What do you think?

Reply:
sure, if we're planning for a one off approach then i guess this is ok; but we do need to ensure that we have proper time allocated for said migration so that we don't have small + incremental pile ups. i'm just worried that the migration won't happen so soon but we keep making allowances like this, which eventually makes the migration have a bigger scope than expectedc

Reply:
Yeah I understand your concern and thanks for sharing it. That's why we need to build a proper backlog to keep track of these issues and plan our capacity to do the migration. We will work on it and plan it together in the near future.